src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1883,7 +1883,7 @@ jint Arguments::set_aggressive_heap_flags() {
     }
   }
 
-#if !defined(_ALLBSD_SOURCE) && !defined(AIX)  // UseLargePages is not yet supported on BSD and AIX.
+#if !defined(_ALLBSD_SOURCE) && !defined(__FreeBSD__) && !defined(AIX)  // UseLargePages is not yet supported on most of BSDs and AIX.
```

Comment:
The logic here looks incorrect to me.  Compare the change you made on line 3852.  Previously the conditional here was the logical inverse of the conditional there.  It now isn't.  The simplest way to keep the two conditionals in sync may be to simply make one the direct inverse of the other using !() around it.

My understanding here is that the intent of the change is that UseLargePages now gets set to true on FreeBSD, but not on other BSD.

Reply:
good catch indeed.